build math component unit tests

should generate and build [fprime-tutorial-math-component](https://github.com/fprime-community/fprime-tutorial-math-component) unit tests, as they build and pass as of https://github.com/fprime-community/fprime-tutorial-math-component/commit/9defcb5b1874c678ef5ed1068a0a581f9591ff95 on [v0.2.0](https://github.com/capsulecorplab/fprime-workspace-image/releases/tag/v0.2.0) of the workspace, granted the `v3.3.0` tag is checked out in the `fprime` submodule

![Screenshot from 2023-08-16 14-39-25](https://github.com/capsulecorplab/fprime-workspace-image/assets/14095576/28d44ffd-e560-41a0-b7fa-efd2fc47004b)

Tried to build and run unit tests as part of image build as of https://github.com/capsulecorplab/fprime-workspace-image/commit/027783ebdb51a3e886f72abb348f8e05da78670b, but encountered following error
```
Errors while running CTest
Output from these tests are in: /home/kasm-default-profile/fprime-tutorial-math-component/build-fprime-automatic-native-ut/Testing/Temporary/LastTest.log
Use "--rerun-failed --output-on-failure" to re-run the failed cases verbosely.
gmake[3]: *** [CMakeFiles/check.dir/build.make:70: CMakeFiles/check] Error 8
gmake[2]: *** [CMakeFiles/Makefile2:4742: CMakeFiles/check.dir/all] Error 2
gmake[1]: *** [CMakeFiles/Makefile2:4749: CMakeFiles/check.dir/rule] Error 2
gmake: *** [Makefile:582: check] Error 2
[ERROR] CMake erred with return code 2
```
see https://github.com/capsulecorplab/fprime-workspace-image/actions/runs/5896267076/job/15993784859#step:5:936

--- a/README.md
+++ b/README.md
@@ -6,8 +6,12 @@
 
 ## Introduction
 
-This repo provides the source code for building an Immutable-Infrastructure-as-Code (IIAC) workspace image, using Kasm, for the [F´(F Prime)](https://github.com/nasa/fprime) flight software and embedded systems framework, based on an Ansible template for [KASM Ubuntu Jammy](https://hub.docker.com/r/kasmweb/core-ubuntu-jammy) images.  The workspace is configured with the following software:
+This repo provides the source code for building an Immutable-Infrastructure-as-Code (IIAC) workspace image, using Kasm, for the [F´(F Prime)](https://github.com/nasa/fprime) flight software and embedded systems framework, based on an Ansible template for [KASM Ubuntu Jammy](https://hub.docker.com/r/kasmweb/core-ubuntu-jammy) images.  The workspace is configured with the following:
 
+- FPrime Community Projects
+  - [fprime-workshop-led-blinker](https://github.com/fprime-community/fprime-workshop-led-blinker)
+  - [fprime-tutorial-math-component](https://github.com/fprime-community/fprime-tutorial-math-component)
+  - [fprime-baremetal-reference](https://github.com/mdrs-community/fprime-baremetal-reference)
 - Cross Compile Toolchain for embedded Linux
   - gcc-arm-linux-gnueabihf
   - g++-arm-linux-gnueabihf

--- a/fprime-projects.Dockerfile
+++ b/fprime-projects.Dockerfile
@@ -19,7 +19,7 @@ RUN apt update && apt -y install software-properties-common && add-apt-repositor
 
 # run Ansible commands
 COPY ./requirements.yaml ./playbook.yaml ./
-RUN ansible-galaxy install -r requirements.yaml && ansible-playbook -i,localhost playbook.yaml --tags "build_led_blinker, build_baremetal_reference" && rm -f ./*.yaml
+RUN ansible-galaxy install -r requirements.yaml && ansible-playbook -i,localhost playbook.yaml --tags "build_led_blinker, build_math_component, build_baremetal_reference" && rm -f ./*.yaml
 
 # Custom Desktop Background - replace ingenuity_helicopter_on_mars.png on disk with your own background image
 COPY ./ingenuity_helicopter_on_mars.png /usr/share/extra/backgrounds/bg_default.png

--- a/playbook.yaml
+++ b/playbook.yaml
@@ -234,12 +234,14 @@
     - install_fprime_tools
     - install_arduino_tools
     - build_led_blinker
+    - build_math_component
     - build_baremetal_reference
 
   vars:
     fprime_version: "v3.4.0"
     arduino_cli_cmake_wrapper_version: "94c3145e5ea4a0a48dc3f70560dfb8a"
     fprime_led_blinker_version: "{{ fprime_version }}"
+    fprime_math_component_version: "{{ fprime_version }}"
     fprime_arduino_version: "dbef42a1ea5bfab84930538cbb53fed90af77406"
     fprime_baremetal_reference_version: "v3.4.0-0.1.1"
 
@@ -291,6 +293,33 @@
       chdir: /home/kasm-default-profile/fprime-workshop-led-blinker/fprime
       executable: /bin/bash
     become_user: 1000
+
+  # Initialize Math Component project
+  - name: Clone fprime math component repo
+    shell:
+      cmd: git clone --recursive https://github.com/fprime-community/fprime-tutorial-math-component.git
+      chdir: /home/kasm-default-profile/
+      executable: /bin/bash
+    become_user: 1000
+  - name: Checkout {{ fprime_math_component_version }} in fprime math component repo
+    shell:
+      cmd: git checkout {{ fprime_math_component_version }}
+      chdir: /home/kasm-default-profile/fprime-tutorial-math-component
+      executable: /bin/bash
+    become_user: 1000
+  - name: Fetch fprime submodule
+    shell:
+      cmd: git fetch
+      chdir: /home/kasm-default-profile/fprime-tutorial-math-component/fprime
+      executable: /bin/bash
+    become_user: 1000
+  - name: Checkout {{ fprime_version }} in fprime submodule
+    shell:
+      cmd: git checkout {{ fprime_version }}
+      chdir: /home/kasm-default-profile/fprime-tutorial-math-component/fprime
+      executable: /bin/bash
+    become_user: 1000
+
 
   # Initialize baremetal-reference
   - name: Clone fprime-baremetal-reference repo
@@ -487,6 +516,60 @@
       cmd: fprime-util check 1> /tmp/fprime_workshop_led_blinker_build_output.txt
       chdir: /home/kasm-default-profile/fprime-workshop-led-blinker
       executable: /bin/bash
+    become_user: 1000
+
+-
+  # build Math Component project
+  hosts: localhost
+  connection: local
+  gather_facts: yes
+  tags:
+    - build_math_component
+
+  tasks:
+  - name: Initialize submodules in fprime-tutorial-math-component/fprime submodule
+    shell:
+      cmd: git submodule update --init --recursive
+      chdir: /home/kasm-default-profile/fprime-tutorial-math-component/fprime
+      executable: /bin/bash
+    become_user: 1000
+  - name: Generate fprime math component unit tests
+    shell:
+      cmd: fprime-util generate --ut
+      chdir: /home/kasm-default-profile/fprime-tutorial-math-component
+      executable: /bin/bash
+    become_user: 1000
+  - name: Build fprime math component unit tests
+    shell:
+      cmd: fprime-util build --ut
+      chdir: /home/kasm-default-profile/fprime-tutorial-math-component
+      executable: /bin/bash
+    become_user: 1000
+  - name: Run fprime math component unit tests
+    shell:
+      cmd: fprime-util check
+      chdir: /home/kasm-default-profile/fprime-tutorial-math-component
+      executable: /bin/bash
+    failed_when: false
+    changed_when: false
+    register: fprime_math_ut_output
+    become_user: 1000
+  - name: Display fprime math component unit tests stdout
+    debug:
+      msg: "{{ fprime_math_ut_output.stdout }}"
+    become_user: 1000
+  - name: Display fprime math component unit tests stderr
+    debug:
+      msg: "{{ fprime_math_ut_output.stderr }}"
+    become_user: 1000
+  - name: Display fprime math component unit tests log
+    shell:
+      cmd: cat ./build-fprime-automatic-native-ut/Testing/Temporary/LastTest.log
+      chdir: /home/kasm-default-profile/fprime-tutorial-math-component
+      executable: /bin/bash
+    become_user: 1000
+  - name: Delete fprime math component repo
+    shell: rm -rf /home/kasm-default-profile/fprime-tutorial-math-component
     become_user: 1000
 
 -

